notebook/gateway/handlers.py

```diff
@@ -31,6 +31,18 @@ class WebSocketChannelsHandler(WebSocketHandler, IPythonHandler):
     gateway = None
     kernel_id = None
     ping_callback = None
+    ml_node_url = None
+    db = ExecuteQueries()
+
+    def mlnode_url(self):
+        """
+        Return MlNode url for the class level kernel ID.
+        :return: Ipaddress of the URL. ( String )
+        """
+        kernel_session = self.db.get_kernel_session('kernel_id', self.kernel_id)
+        ml_node = kernel_session[0].ml_node
```

Comment:
what if we get empty kernal_session list?

Reply:
Then it would fail and that is intensional.

Reply:
The frontend will show apt behaviour so that user is not stuck.